src/extension.ts

```diff
@@ -121,12 +120,24 @@ async function getPRsPerSHAs(){
 				if (issuesBySHAs.length === 0) {
 					vscode.window.showErrorMessage("No search results. Try selecting a bigger piece of code or another file.");
 				} else {
-					issuesBySHAs.forEach((issue: { url: any; }) => {
+					issuesBySHAs.forEach(async (issue: { url: any; }) => {
 						const issueUrl = issue.url;
 	
-						octokit.request(`GET ${issueUrl}/comments`).then(octoresp => {
+						await octokit.request(`GET ${issueUrl}/comments`).then(async octoresp => {
 							// this paints the panel
-							watermelonPanel.currentPanel?.doRefactor({ command: "prs", data: octoresp.data})
+							octoresp.data.forEach(async (issue: { issue_url: any, title: string }) => {
+								const issueUrl = issue.issue_url;
+
+								await octokit.request(`GET ${issueUrl}`).then(octoresp2 => {
```

Comment:
Should we use GraphQL?
This seems very complicated... There are 3 requeests
https://docs.github.com/en/graphql/reference/objects#issue

Reply:
A single call would contain all in an organized fashion, instead of 3 requests as a waterfall 
https://docs.github.com/en/graphql/reference/objects#pullrequest